### ISSUE TITLE
Reqesting Error (Fuzzy Crafting)

### ISSUE THREAD

**Issue by Michal-MK** [RS485/LogisticsPipes#712]
Buildracft 7.0.9
Logistics Pipes 0.10.0.5

Server - when I request Forestry bronze from the system and the system has to craft it the server Disconects me. I used normal crafting table with a MK1 Crafting pipe with a fuzzy upgrade that is set to use OreDict.
Client log doesn't say anything other than: A fatal error has occured, this connection is terminated
Server gives this error: http://pastebin.com/4Rc7UNfc

**Comment by SonOfTheStars** [comment]
Fuzzy Crafting is currently kind of Broken
But ive never managed to run into this kind of IAE
 Cause is:
  java.lang.IllegalArgumentException: zero count needed, promises not needed.
Wich causes the packet not to be handled and your Connection to get Terminated.

@davboecki I summon thee!

**Comment by saulrh** [comment]
I was getting this last night. Tried to update the LP in my FTB Infinity (logisticspipes-0.8.3.122 -> logisticspipes-0.9.1.132, Forge 10.13.3.1408 and BuildCraft 6.4.16) and got this any time a request failed. I didn't spend much time debugging it; it was sufficiently reproducible and crippling that I just reverted. If it'd be helpful I could probably try again and provide an error log.

**Comment by NitroxydeX** [comment]
Checking in to this. It seems that it's turning the server to freeze-state through the Server thread.
Here some nice looking stall-report:

```
"Server thread" #34 prio=5
RUNNABLE
    at logisticspipes.utils.item.ItemIdentifier.get(ItemIdentifier.java:340)
    at logisticspipes.utils.InventoryUtil.itemCount(InventoryUtil.java:45)
    at logisticspipes.pipes.PipeItemsProviderLogistics.getTotalItemCount(PipeItemsProviderLogistics.java:134)
    at logisticspipes.pipes.PipeItemsProviderLogistics.getAvailableItemCount(PipeItemsProviderLogistics.java:244)
    at logisticspipes.pipes.PipeItemsProviderLogistics.canProvide(PipeItemsProviderLogistics.java:296)
    at logisticspipes.request.RequestTreeNode.checkProvider(RequestTreeNode.java:288)
    at logisticspipes.request.RequestTreeNode.<init>(RequestTreeNode.java:57)
    at logisticspipes.request.RequestTreeNode.getSubRequests(RequestTreeNode.java:609)
    at logisticspipes.request.RequestTreeNode.access$100(RequestTreeNode.java:37)
    at logisticspipes.request.RequestTreeNode$CraftingSorterNode.calculateMaxWork(RequestTreeNode.java:536)
    at logisticspipes.request.RequestTreeNode$CraftingSorterNode.addWorkPromisesToTree(RequestTreeNode.java:553)
    at logisticspipes.request.RequestTreeNode.checkCrafting(RequestTreeNode.java:482)
    at logisticspipes.request.RequestTreeNode.<init>(RequestTreeNode.java:65)
    at logisticspipes.request.RequestTreeNode.<init>(RequestTreeNode.java:40)
    at logisticspipes.request.RequestTree.<init>(RequestTree.java:43)
    at logisticspipes.request.RequestTree.request(RequestTree.java:177)
    at logisticspipes.request.RequestTree.requestPartial(RequestTree.java:206)
    at logisticspipes.modules.ModuleCrafter.tick(ModuleCrafter.java:253)
    at logisticspipes.pipes.basic.CoreRoutedPipe.updateEntity(CoreRoutedPipe.java:468)
    at logisticspipes.pipes.basic.LogisticsTileGenericPipe.func_145845_h(LogisticsTileGenericPipe.java:205)
    at net.minecraft.world.World.func_72939_s(World.java:1939)
    at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:489)
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:636)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:334)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:396)
    at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685)
```

Running BC 7.0.10 with LP 0.10.0.5

**Comment by davboecki** [comment]
@NitroxydeX What is that, how did you get it, what is it supposed to show?
What you posted points to this line: https://github.com/RS485/LogisticsPipes/blob/dev/common/logisticspipes/utils/item/ItemIdentifier.java#L340 which doesn't help at all.

**Comment by NitroxydeX** [comment]
@davboecki I get this via Sampler Stall-Report if the server freezes for more than 30000ms it generates a stall report with the Server thread that it's causing. 

I think it's a similar crash with requesting items somehow.

**Comment by bziemons** [comment]
About the hang: please open a new issue if that still happens with the newest changes. The error was fixed in ce7e4d7e0a03e03f9a527371f70d2d55764df339.
